packages/grid/data-grid/src/tests/layout.DataGrid.test.tsx

```diff
@@ -642,22 +641,13 @@ describe('<DataGrid /> - Layout & Warnings', () => {
   });
 
   describe('warnings', () => {
-    before(() => {
-      PropTypes.resetWarningCache();
-    });
```

Comment:
Seeing this diff made me suspicious, so I gave it a try to confirm it: When running the tests in watch mode, the first run is green, but the second one is red. What motived the change of approach?

Reply:
To test the warnings we were using `PropTypes.checkPropTypes` and feeding it with the prop-types. However, to get the prop-types they had to be attached to the exported component, which is a memo. Memoized components can't have prop-types so the generator attaches them to the `DataGridRaw`.

https://github.com/mui-org/material-ui-x/blob/2cd18f9cc5f4723a20f4ddec2ef3864e398e726b/packages/grid/data-grid/src/DataGrid.tsx#L38-L41

Since there's no more way to access the prop-types directly, I had to render the component to be able to test. The reason for seeing the second run red in watch mode: https://github.com/facebook/react/issues/18251#issuecomment-607567390